### ISSUE TITLE
"rsyslogd -N <level>" should exit with non-zero exit code when configuration is invalid

### ISSUE THREAD

**Issue by Whissi** [rsyslog/rsyslog#79]
Hi,

1. Make an error in your rsyslog configuration file.

2. No run ```rsyslogd``` in *config check mode*, ```rsyslogd -N 999 -f /etc/rsyslog.conf```

You will see the detected error, but rsyslogd will end with exit code 0.

Tested with v7.6.3.

PS: I was wondering because I tested this feature before and it was working, I got an non-zero exit code. But now, while testing another configuration, it didn't... Seems like it is currently only working for really fatal errors:
```
*.=debug;\
auth,authpriv.none;\
news.none;mail.none action(
	type="omfile"
	File="/var/log/debug"
)
```
which will result in
```
rsyslogd: version 7.6.3, config validation run (level 999), master config /etc/rsyslog.conf
rsyslogd: error during parsing file /etc/rsyslog.conf, on or before line 13: invalid character '(' in object definition - is there an invalid escape sequence somewhere? [try http://www.rsyslog.com/e/2207 ]
rsyslogd: error during parsing file /etc/rsyslog.conf, on or before line 13: invalid character '1' in object definition - is there an invalid escape sequence somewhere? [try http://www.rsyslog.com/e/2207 ]
rsyslogd: error during parsing file /etc/rsyslog.conf, on or before line 13: invalid character '4' in object definition - is there an invalid escape sequence somewhere? [try http://www.rsyslog.com/e/2207 ]
rsyslogd: error during parsing file /etc/rsyslog.conf, on or before line 13: syntax error on token ')' [try http://www.rsyslog.com/e/2207 ]
rsyslogd: CONFIG ERROR: could not interpret master config file '/etc/rsyslog.conf'. [try http://www.rsyslog.com/e/2207 ]
rsyslogd: run failed with error -2207 (see rsyslog.h or try http://www.rsyslog.com/e/2207 to learn what that number means)
```
output.

But it would be nice if ```rsyslogd``` would always exit with a non-zero exit code when there's an error in the configuration when running in config test mode (```-N````). That's how other daemon's config test mode behaves (see nginx, apache2).

**Comment by moander** [comment]
Same issue still on rsyslogd 8.7.0

**Comment by michalsuba** [comment]
+1

**Comment by rgerhards** [comment]
This is more or less a duplicate of https://github.com/rsyslog/rsyslog/issues/78 which is being worked on.

**Comment by rgerhards** [comment]
fix to be released with 8.10.0.
